clippy_lints/src/methods/mod.rs

```diff
@@ -1961,6 +1962,56 @@ declare_clippy_lint! {
     "replace `.iter().count()` with `.len()`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for the usage of `_.to_owned()`, on a `Cow<'_, _>`.
+    ///
+    /// ### Why is this bad?
+    /// `Cow`s can contain either `Borrowed` data or `Owned` data.
+    /// The `into_owned()` method on a `Cow` can be used to clone the
+    /// underlying data and become a `Cow::Owned`. Calling `to_owned()`,
+    /// on the other hand, clones the `Cow` itself, rather than the
+    /// contained data.
+    /// Given the similar naming, the use of `to_owned()` is suspicious
+    /// and can potentially lead to undefined behavior if it's
+    /// inadvertently used on data coming from unsafe contexts.
+    /// Consider replacing `to_owned` with `into_owned` if the desired
+    /// result is getting a `Cow::Owned`, or explitly calling `clone()`
+    /// if the desired result is cloning the `Cow`.
```

Comment:
I think this still places too much emphasis on `into_owned`, given that in our checks the suggestion would lead to errors. Perhaps describing the behavior of `to_owned` and why it may be surprising, and then giving both alternatives (`clone` vs. `into_owned`) would help.